packages/@aws-cdk/aws-kinesisfirehose/lib/delivery-stream.ts

```diff
@@ -0,0 +1,256 @@
+   */
+  readonly role?: iam.IRole;
+}
+
+/**
+ * Create a Kinesis Data Firehose delivery stream
+ *
+ * @resource AWS::KinesisFirehose::DeliveryStream
+ */
+export class DeliveryStream extends DeliveryStreamBase {
+  /**
+   * Import an existing delivery stream from its name.
+   */
+  static fromDeliveryStreamName(scope: Construct, id: string, deliveryStreamName: string): IDeliveryStream {
+    return this.fromDeliveryStreamAttributes(scope, id, { deliveryStreamName });
+  }
+
+  /**
+   * Import an existing delivery stream from its ARN.
+   */
+  static fromDeliveryStreamArn(scope: Construct, id: string, deliveryStreamArn: string): IDeliveryStream {
+    return this.fromDeliveryStreamAttributes(scope, id, { deliveryStreamArn });
+  }
+
+  /**
+   * Import an existing delivery stream from its attributes.
+   */
+  static fromDeliveryStreamAttributes(scope: Construct, id: string, attrs: DeliveryStreamAttributes): IDeliveryStream {
+    if (!attrs.deliveryStreamName && !attrs.deliveryStreamArn) {
+      throw new Error('Either deliveryStreamName or deliveryStreamArn must be provided in DeliveryStreamAttributes');
+    }
+    const deliveryStreamName = attrs.deliveryStreamName ?? cdk.Stack.of(scope).parseArn(attrs.deliveryStreamArn!).resourceName;
+    if (!deliveryStreamName) {
+      throw new Error(`Could not import delivery stream from malformatted ARN ${attrs.deliveryStreamArn}: could not determine resource name`);
+    }
+    const deliveryStreamArn = attrs.deliveryStreamArn ?? cdk.Stack.of(scope).formatArn({
+      service: 'firehose',
+      resource: 'deliverystream',
+      resourceName: attrs.deliveryStreamName,
+    });
+    class Import extends DeliveryStreamBase {
+      public readonly deliveryStreamName = deliveryStreamName!;
+      public readonly deliveryStreamArn = deliveryStreamArn;
+      public readonly grantPrincipal = attrs.role ?? new iam.UnknownPrincipal({ resource: this });
+    }
+    return new Import(scope, id);
+  }
+
+  readonly deliveryStreamName: string;
+
+  readonly deliveryStreamArn: string;
+
+  readonly grantPrincipal: iam.IPrincipal;
+
+  constructor(scope: Construct, id: string, props: DeliveryStreamProps) {
+    super(scope, id);
+
+    const role = props.role ?? new iam.Role(this, 'Service Role', {
+      assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+    });
+    this.grantPrincipal = role;
+
+    if (props.destinations.length !== 1) {
+      throw new Error(`Only one destination is allowed per delivery stream, given ${props.destinations.length}`);
+    }
+    const destinationConfig = props.destinations[0].bind(this, { deliveryStream: this });
+
+    const resource = new CfnDeliveryStream(this, 'Resource', {
+      deliveryStreamName: props.deliveryStreamName,
+      deliveryStreamType: 'DirectPut',
+      ...destinationConfig.properties,
+    });
+    resource.node.addDependency(this.grantPrincipal);
```

Comment:
This probably deserves a comment why it's here.

Reply:
This is also probably not what you want. You're probably looking to take a dependency on a *particular* grant, not on all grants every done to this principal. The grants have an API for doing exactly that.

Reply:
I removed this line because it's not needed until we add Kinesis streams as a source or server-side encryption. However, it's a bit tedious to have to write `Node.of(scope).addDependency(grant)` every time I have to give permissions to the role (especially since `Grant.applyBefore` uses core.IConstruct instead of constructs.IConstruct). Makes it seem like depending on all of the grants in the scope a lot easier

Reply:
Implemented a similar sort of dependency on grants for the destination permissions in a307108, but it's kind of an awkward "threading-through". It's surprising to me that I don't see another obvious example of this requirement (ensuring a CFN resource deploys after a bunch of grants created all over the place), especially in another integration module.